src/System.Net.Http/tests/FunctionalTests/HttpClientHandlerTest.cs

```diff
@@ -57,5 +83,142 @@ public void SendAsync_SimpleGet_Success()
                     "Expected cancellation requested on token attached to exception.");
             }
         }
+
+        [Fact]
+        public void GetAsyncAutomaticDecompression()
+        {
+            var handler = new HttpClientHandler();
+            if (!handler.SupportsAutomaticDecompression)
+            {
+                return;
+            }
+
+            var client = new HttpClient(handler);
+            var response = client.GetAsync(HttpTestServers.RemoteServerGzipUri).Result;
+            Assert.Equal(HttpStatusCode.OK, response.StatusCode);
+            var responseContent = response.Content.ReadAsStringAsync().Result;
+            Assert.True(responseContent.Contains(DecompressedContentPart));
+            handler.Dispose();
+            client.Dispose();
+        }
+
+        [Fact]
+        public void GetAsyncBasicAuthentication()
+        {
+            var uri = HttpTestServers.BasicAuthUriForCreds(Username, Password);
+            var unAuthClient = new HttpClient();
+            var unAuthResponse = unAuthClient.GetAsync(uri).Result;
+            Assert.Equal(HttpStatusCode.Unauthorized, unAuthResponse.StatusCode);
+            unAuthClient.Dispose();
+
+            var handler = new HttpClientHandler();
+            handler.Credentials = _credential;
+            var client = new HttpClient(handler);
+            var response = client.GetAsync(uri).Result;
+            Assert.Equal(HttpStatusCode.OK, response.StatusCode);
+            Assert.Equal(new Version(1,1), response.Version);
+            handler.Dispose();
+            client.Dispose();
+        }
+
+        [Fact]
+        public void UriRedirectionDefault()
+        {
+            var handler = new HttpClientHandler();
+            Assert.True(handler.AllowAutoRedirect);
+            handler.AllowAutoRedirect = false;
+            var client = new HttpClient(handler);
+            var response = client.GetAsync(HttpTestServers.RemoteServerRedirectUri).Result;
+            Assert.Equal(HttpStatusCode.Redirect, response.StatusCode);
+            client.Dispose();
+            handler.Dispose();
+        }
+
+        [Fact]
+        public void UriRedirectionAllow()
+        {
+            var handler = new HttpClientHandler();
+            var client = new HttpClient(handler);
+            var response = client.GetAsync(HttpTestServers.RemoteServerRedirectUri).Result;
+            Assert.Equal(HttpStatusCode.OK, response.StatusCode);
+            handler.Dispose();
+            client.Dispose();
+        }
+
+        [Theory]
+        [InlineData(6)]
+        public void UriMaxRedirectionAllow(int hops)
+        {
+            var handler = new HttpClientHandler();
+            handler.MaxAutomaticRedirections = hops;
+            var client = new HttpClient(handler);
+            Assert.Throws<System.AggregateException>(() =>
+                { var _ = client.GetAsync(HttpTestServers.RedirectUriHops(hops + 1)).Result; }
```

Comment:
By using await as I suggested earlier, you can change this Throws to catch the actual exception type that's getting thrown (before it gets wrapped in an aggregate).